java/ql/src/semmle/code/java/frameworks/spring/SpringWebClient.qll

```diff
@@ -27,3 +27,116 @@ class SpringWebClient extends Interface {
     this.hasQualifiedName("org.springframework.web.reactive.function.client", "WebClient")
   }
 }
+
+/**
+ * An abstract class representing all Spring Rest Template methods
+ * which take an URL as an argument.
+ */
+abstract class SpringRestTemplateUrlMethods extends Method {
+  /** Gets the argument which corresponds to a URL */
+  abstract Argument getUrlArgument(MethodAccess ma);
+}
```

Comment:
Lots of issues here. First of, generally avoid public `abstract` classes unless you really want to expose their open-ended semantics. In this case this class and all of its subclasses should be collapsed to just a single class.
Second, is a list of method names the best way to capture this? Or is it better to actually model things as the qldoc describes? I.e. a method on `SpringRestTemplate` with a parameter of type `TypeUri` or `TypeUrl`?
Once we've rewritten the code to that point it makes sense to ask whether this then belongs in the Spring framework library or whether it should be closer to the query (I think the latter is more appropriate if we simply model the methods based on parameter type).

Reply:
@aschackmull The arguments may or may not be `java.net.URL`'s. I have changed the doc to reflect the same. 
As for the refactoring. I have inlined all classes into a single class and moved it to the `RequestForgeryCustomizations.qll` file. I think it looks much better this way now.